tools/markdown-to-html/index.bzl

```diff
@@ -0,0 +1,48 @@
+"""
+  Implementation of the "markdown_to_html" rule. The implementation runs the transform
+  executable in order to create the outputs for the specified source files.
+"""
+def _markdown_to_html(ctx):
+  input_files = ctx.files.srcs;
+  args = ctx.actions.args()
+  expected_outputs = [];
+
+  for input_file in input_files:
+    basename = input_file.basename.replace('.md', '')
+    output_file = ctx.actions.declare_file("%s.html" % basename)
+    expected_outputs += [output_file]
+
+    # Add the input file and it's related output to the arguments that
+    # will be passed to the transformer executable.
+    args.add("%s=%s" % (input_file.path, output_file.path))
```

Comment:
We need to pass the output path because the input paths differ from the output paths.


We could probably do better by just passing the `bazel-bin` directory once, and just replacing the extension when writing. Updated. good idea.